src/main/c/ssl.c

```diff
@@ -1178,11 +1178,11 @@ TCN_IMPLEMENT_CALL(jint, SSL, getLastErrorNumber)(TCN_STDARGS) {
 }
 
 static void ssl_info_callback(const SSL *ssl, int where, int ret) {
-    tcn_ssl_ctxt_t *c;
+    uint16_t *handshakeCount = NULL;
     if (0 != (where & SSL_CB_HANDSHAKE_START)) {
-        c = SSL_get_app_data2(ssl);
-        if (c != NULL) {
-            ++c->handshakeCount;
+        handshakeCount = (uint16_t*) SSL_get_app_data3(ssl);
+        if (handshakeCount != NULL) {
+            ++(*handshakeCount);
```

Comment:
@normanmaurer - You would be fine but I vote leave it. Prefix inc and dereference are in the same precedence group, but parsing is from right to left...but I think what you have here prevents people from having to double check operator precedence every time they look at it ;)